src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -193,7 +188,11 @@ const HeatmapPlot = (props) => {
     );
   }
 
-  if (isHeatmapGenesLoading) {
+  if (
+    isHeatmapGenesLoading
+    || cellSetsLoading
+    || cellSetsHierarchy.length === 0
```

Comment:
this doesn't sound right: if `cellSetsHierarchy` is 0, then we shouldn't be in loading state, then we will just have an empty component

Reply:
Oh, I see what you mean. I thought we're handling the case where cellSets is empty while waiting for the cell sets to load. So this is to handle the case where cell sets is empty, even after cell sets are loaded?

Reply:
I don't understand this comment Agi. I don't see how `cellSetsHierarchy` should have any impact on the decision whether to be in a loading state or not. Wouldn't just `cellSetsLoading` be enough?

Reply:
maybe we should have a short chat about this, maybe we are talking about different things here